app/models/bus_stop.rb

```diff
@@ -182,9 +182,17 @@ def self.to_csv(limited_attributes: false)
     end
   end
 
+  def who_completed
+    User.find(completed_by).try :name
+  end
+
   private
 
-  def assign_completion_timestamp
-    assign_attributes completed_at: (completed? ? DateTime.current : nil)
+  def assign_completion_attributes
+    whodunnit = versions.last.whodunnit
+    assign_attributes(
```

Comment:
I think it reads more easily with the conditional a level up, and using setters rather than `assign_attributes`:

```ruby
if completed?
  completed_at = DateTime.current
  completed_by = versions.last.whodunnit
else
  completed_at = nil
  completed_by = nil
end
```

Reply:
It seemed cleaner the other way but I've always enjoyed putting ternary operators everywhere I could. I toyed with this idea though and feel neutrally. Will change.

I'll see about adding the relationship.

Reply:
Ooh this is a gotcha:
`completed_at = DateTime.current`
- Assigns a local variable with the value DateTime.current.
`self.completed_at = DateTime.current`
- Does the thing you want

Reply:
If so, then that's a serious bug in Rails. Setter methods are a 100% valid way of setting attributes in Rails.

Reply:
Personally I like the ternaries, I think it looks cleaner. I know @mboneil10 would disagree though!